components/ai_chat/resources/page/state/ai_chat_context.tsx

```diff
@@ -77,63 +66,10 @@ export function AIChatContextProvider(props: React.PropsWithChildren<Props>) {
   }
 
   React.useEffect(() => {
-    const { Service, Observer } = getAPI()
-    async function initialize() {
-      const [
-        { conversations: visibleConversations },
-        { actionList: allActions },
-        { canShow: canShowPremiumPrompt }
-      ] = await Promise.all([
-        Service.getVisibleConversations(),
-        Service.getActionMenuList(),
-        Service.getCanShowPremiumPrompt()
-      ])
-      setPartialContext({
-        visibleConversations,
-        allActions,
-        canShowPremiumPrompt
-      })
-    }
-
-    async function updateCurrentPremiumStatus() {
-      const { status } = await getAPI().Service.getPremiumStatus()
-      setPartialContext({
-        isPremiumStatusFetching: false,
-        isPremiumUser: (status !== undefined && status !== mojom.PremiumStatus.Inactive),
-        isPremiumUserDisconnected: status === mojom.PremiumStatus.ActiveDisconnected
-      })
-    }
-
-    initialize()
-    updateCurrentPremiumStatus()
-
-    if (context.isHistoryEnabled) {
-      Observer.onConversationListChanged.addListener(
-        (conversations: mojom.Conversation[]) => {
-          setPartialContext({
-            visibleConversations: conversations
-          })
-        }
-      )
-    }
+    setPartialContext(getAPI().UIState)
 
```

Comment:
nit: capital `U` `UIState` reads slightly weird to me - normally that'd be the name of a class but here its a bag of data